opentelemetry-sdk/src/opentelemetry/sdk/trace/export/__init__.py

```diff
@@ -199,16 +201,16 @@ def export(self) -> None:
             else:
                 self.spans_list[idx] = span
                 idx += 1
-        with Context.use(suppress_instrumentation=True):
-            try:
-                # Ignore type b/c the Optional[None]+slicing is too "clever"
-                # for mypy
-                self.span_exporter.export(
-                    self.spans_list[:idx]
-                )  # type: ignore
-            # pylint: disable=broad-except
-            except Exception:
-                logger.exception("Exception while exporting Span batch.")
+        backup_context = get_current()
+        set_current(set_value("suppress_instrumentation", True))
```

Comment:
I know you didn't add this line, but do you know why we have to suppress instrumentation here after we already did it in `on_end`?

Reply:
It's the batch exporter, `export` is executed in a different thread that `on_end`. I think the `on_end` could be removed (in another PR), having this before `export` is called in both, simple and batch span processors, should be enough.

Reply:
@mauriciovasquezbernal would you mind opening an issue to track this?

Reply:
@c24t, @codeboten I checked up the implementation it is actually right. 
`suppress_instrumentation` has two be set before exporting the span in the simple and batch span processors. The `on_end` that @c24t mentioned is the hook on the `SimpleExportSpanProcessor`, and the line of this comment is on the `BatchExportSpanProcessor`, there is not a duplicated logic.

Reply:
Got it, thanks for checking!